### PR TITLE
(PCP-227) Acceptance test of pxp-module-puppet run with 'changed' result

### PR DESCRIPTION
**Pull request by james-stocks**
Use pxp-module-puppet to run puppet with the expected result being 'changed'

[skip ci]

### CODE REVIEW

File: acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      target_identities << "pcp://client0#{i+1}.example.com/agent"
+    end
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--server', "'#{master}'",
+                                                              '--onetime',
+                                                              '--no-daemonize',
+                                                              "--environment", "'#{ENVIRONMENT_NAME}'"]
```
Comment (richardc) [inline acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb]: `"'#{ENVIRONMENT_NAME}'"` is a lot of overquoting.  Just ENVIRONMENT_NAME should work.

File: acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  step "Send an rpc_blocking_request to all agents" do
+    target_identities = []
+    agents.each_with_index do |agent, i|
+      target_identities << "pcp://client0#{i+1}.example.com/agent"
+    end
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--server', "'#{master}'",
```
Comment (richardc) [inline acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb]: Same overquoting of master here.

File: acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                                                              '--onetime',
+                                                              '--no-daemonize',
```
Comment (richardc) [inline acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb]: I believe these flags are there by default.  https://github.com/puppetlabs/pxp-agent/blob/master/modules/pxp-module-puppet#L242

File: acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+test_name 'C93062 - Run puppet and expect \'changed\' result' do
+
+  step 'On master, create a new environment that will result in Changed' do
+    environmentpath = master.puppet['environmentpath']
+    site_manifest = "#{environmentpath}/#{ENVIRONMENT_NAME}/manifests/site.pp"
+    on(master, "cp -r #{environmentpath}/production #{environmentpath}/#{ENVIRONMENT_NAME}")
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  include #{MODULE_NAME}
```
Comment (richardc) [inline acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb]: This use of a module seems overkill.  The entire site.pp could be just the notify resource.

**Comment by Andrew-Hayes** [comment]
Looks good, however is a few i+1 around. perhaps a variable would be better?

agent_index = i+1

**Comment by james-stocks** [comment]
@Andrew-Hayes : agreed - but that pattern is used throughout the repo; so should be improved everywhere and not just this test.
At the same time; making that improvement is perhaps low-value; as the acceptance suite needs to be updated to use Puppet certs (i.e. the agent certname) instead of the test certs that these i + 1's are referring to; so they'll be removed in the near future anyway.

File: acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
+require 'pxp-agent/test_helper.rb'
+
+ENVIRONMENT_NAME = 'changed_result'
+MODULE_NAME = 'changed'
```
Comment (richardc) [inline acceptance/tests/pxp-module-puppet/run_puppet_result_changed.rb]: I think MODULE_NAME is leftover from a previous swing?   Nothing is using it now.

**Pull request merged**